src/main/java/org/jenkinsci/plugins/docker/commons/impl/ServerKeyMaterialFactory.java

```diff
@@ -81,9 +81,10 @@ public KeyMaterial materialize() throws IOException, InterruptedException {
 
         if (key != null && cert != null && ca != null) {
             final FilePath tempCredsDir = new FilePath(getContext().getBaseDir(), UUID.randomUUID().toString());
+            tempCredsDir.mkdirs();
 
             // protect this information from prying eyes
-            tempCredsDir.chmod(0600);
+            tempCredsDir.chmod(0700);
```

Comment:
Now that the chmod call succeeds, using `0600` causes `DockerServerEndpointTest` to fail because the files in the folder are inaccessible if the executable bit is not set.